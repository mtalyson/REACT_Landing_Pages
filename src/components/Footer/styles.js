import styled, { css } from 'styled-components';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;

    p {
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
