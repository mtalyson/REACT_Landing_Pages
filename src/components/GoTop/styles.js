import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
  `}
`;
