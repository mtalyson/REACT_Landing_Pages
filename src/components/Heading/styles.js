import styled, { css } from 'styled-components';

const titleSize = {
  ssmall: (theme) => css`
    font-size: ${theme.font.sizes.small};
  `,
  small: (theme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.font.sizes.large};
  `,
  big: (theme) => css`
    font-size: ${theme.font.sizes.xlarge};
  `,
  shuge: (theme) => css`
    font-size: ${theme.font.sizes.shuge};
    ${mediaFont(theme)};
  `,
  huge: (theme) => css`
    font-size: ${theme.font.sizes.xhuge};
    ${mediaFont(theme)};
  `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.xlarge};
  }
`;

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
