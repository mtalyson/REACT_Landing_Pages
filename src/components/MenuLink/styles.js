import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: block;
    position: relative;
    text-decoration: none;
    font-size: ${theme.font.sizes.xsmall};
    padding: ${theme.spacings.smedium};
    color: ${theme.colors.primaryColor};

    &::after {
      content: '';
      position: absolute;
      bottom: 1.1rem;
      left: 50%;
      width: 0;
      height: 0.2rem;
      background: ${theme.colors.secondaryColor};
      transition: all 300ms ease-in-out;
    }

    &:hover::after {
      left: 25%;
      width: 50%;
    }
  `}
`;
