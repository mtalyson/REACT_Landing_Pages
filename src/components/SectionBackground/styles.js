import styled, { css } from 'styled-components';

const containerBackgroungActivate = (theme) => css`
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
`;

export const Container = styled.div`
  ${({ theme, background }) => css`
    display: flex;
    align-items: center;

    background: ${theme.colors.white};
    color: ${theme.colors.primaryColor};

    ${background && containerBackgroungActivate(theme)};

    min-height: 100vh;
  `}
`;
