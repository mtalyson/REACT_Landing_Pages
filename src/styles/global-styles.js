import { createGlobalStyle } from 'styled-components';

export const GlobalStyles = createGlobalStyle`
  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }

  html {
    font-size: 62.5%;
    scroll-behavior: smooth;
  }

  body {
    font-size: 1.6rem;
    font-family: ${({ theme }) => theme.font.family.default};
  }

  h1, h2, h3, h4, h5, h6 {
    font-family: ${({ theme }) => theme.font.family.title};
    margin: ${({ theme }) => theme.spacings.large} 0;
    font-weight: 900;
  }

  p {
    margin-bottom: ${({ theme }) => theme.spacings.xmedium};
  }

  ul, ol {
    margin: ${({ theme }) => theme.spacings.medium};
    padding: ${({ theme }) => theme.spacings.medium};
  }

  a {
    color: ${({ theme }) => theme.colors.secondaryColor};
  }

  table {
    min-width: 360px;
    width: 100%;
    border-collapse: collapse;
  }

  table caption {
    font-style: italic;
    font-size: 1.6rem;
    text-align: left;
    margin-bottom: 1rem;
  }

  table td, table th {
    white-space: nowrap;
    padding: 1rem;
    text-align: left;
    border: 0.1rem solid ${({ theme }) => theme.colors.mediumGray};;
  }

  tfoot td, table th {
    background: ${({ theme }) => theme.colors.mediumGray};;
  } 

  .responsive-table {
    width: 100%;
    overflow-y: auto;
  }
`;
