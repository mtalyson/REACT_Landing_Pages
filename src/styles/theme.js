export const theme = {
  colors: {
    primaryColor: '#0A1128',
    secondaryColor: '#dc143c',
    white: '#FFFFFF',
  },
  font: {
    family: {
      default: "'Open Sans', sans-serif;",
      title: "'Montserrat', sans-serif;",
    },
  },
};
