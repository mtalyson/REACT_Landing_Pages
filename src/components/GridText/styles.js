import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
  `}
`;

export const GridItem = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      position: relative;
      left: 5rem;
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.font.sizes.xhuge};
      top: -2.5rem;
      left: -5.5rem;
      transform: rotate(5deg);
    }
  `}
`;
