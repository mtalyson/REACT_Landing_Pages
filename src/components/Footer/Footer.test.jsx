import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { Footer } from '.';

describe('<Footer />', () => {
  it('should render footer', () => {
    const { container } = renderTheme(<Footer footerHtml={'<h1>Olá</h1>'} />);

    expect(screen.getByRole('heading', { name: 'Olá' })).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
