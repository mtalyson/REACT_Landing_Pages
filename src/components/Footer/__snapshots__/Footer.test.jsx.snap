// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render footer 1`] = `
.c0 {
  text-align: center;
}

.c0 p {
  font-size: 1.6rem;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c2 {
  font-size: 2.4rem;
}

<div>
  <footer
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h1>
          Ol√°
        </h1>
      </div>
    </div>
  </footer>
</div>
`;
